src/components/_classes/component/Component.js

```diff
@@ -2975,7 +2975,7 @@ export default class Component extends Element {
     this.calculatedValue = fastCloneDeep(calculatedValue);
 
     if (changed) {
-      if (!flags.noPristineChangeOnModified) {
+      if (!flags.noPristineChangeOnModified && this.root.initialized) {
```

Comment:
Why can't we write tests for this? Even though it's a relatively simple change, a test or two would define the expected behavior of our program for developers and prevent regressions. It's also a good opportunity to test your assumptions to make sure there are no edge cases that haven't been considered.